### ISSUE TITLE
chown very slow

### ISSUE THREAD

**Issue by ben-github** [jupyter/docker-stacks#81]
The docker file for the minimal-notebook runs

```
chown -R $NB_USER:users $CONDA_DIR && \
chown -R $NB_USER:users /home/$NB_USER
```

in the file. I don't know why, but this is very slow. If I build a new docker image and pass a different NB_USER value (so this step isn't cached) it takes ~10 minutes to finish running the chown. This may have to do with how aufs is working behind the scenes (I'm running on Debian, using aufs for docker storage). Since the file is cached on rebuilds, this isn't a big deal, but does the jovyan user need to be owner of /opt/conda? If so, maybe you could create that user first and install conda as user jovyan?

**Comment by parente** [comment]
The chown is there to support the NB_UID option which allows the notebook server to run under a different UID (e.g., in JupyterHub). 

Are you setting that env var? If not, that chown shouldn't be running unless the if check here is busted:

https://github.com/jupyter/docker-stacks/blob/master/minimal-notebook/start-notebook.sh#L4

**Comment by ben-github** [comment]
Yes, I'm setting the env var with -e NB_UID when creating the docker container. The slow chown command is from https://github.com/jupyter/docker-stacks/blob/master/minimal-notebook/Dockerfile not the start-notebook.sh.

It works as expected, it just is very very slow creation of the first docker image and I'm wondering if there is a faster way to get the same features.

**Comment by parente** [comment]
Ah, sorry! Totally forgot about the one in the Dockerfile. Yes, then I too wonder if we can move the user creation up and just install conda as jovyan instead of root.

**Comment by parente** [comment]
Issue fixed but build blocked by #55. The workaround we were using to get R working no longer works.

**Comment by parente** [comment]
Pushed to DockerHub with image tag 2d125a7161b5.
